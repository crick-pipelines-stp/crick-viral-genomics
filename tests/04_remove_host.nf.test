nextflow_pipeline {
    name "Test Pipeline Components"
    script "../main.nf"

    tag "remove_host"

    test("remove_host_illumina") {
        tag "remove_host/illumina"
        when {
            params {
                run_remove_host_reads = true
                run_reporting         = false
                host_genome           = "sacCer3"

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/tests/data/seq_sim/influenza_A/*"
                samplesheet = "$projectDir/tests/data/samplesheets/test_illumina_flu_amp_single.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }

    test("remove_host_nanopore") {
        tag "remove_host/nanopore"
        when {
            params {
                run_remove_host_reads = true
                run_reporting         = false
                host_genome           = "sacCer3"
                run_minimap_align     = true

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/tests/data/seq_sim/influenza_A/*"
                samplesheet = "$projectDir/tests/data/samplesheets/test_illumina_flu_amp_single.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }
}
