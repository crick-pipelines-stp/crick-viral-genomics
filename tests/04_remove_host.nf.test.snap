{
    "remove_host_nanopore": {
        "content": [
            23,
            [
                "00_reference",
                "00_reference/viral_reference.merged.fasta",
                "00_reference/viral_reference.merged.fasta.fai",
                "01_qc",
                "01_qc/mosdepth",
                "01_qc/mosdepth/51-A_Spain_1810_2024_S51.mosdepth.global.dist.txt",
                "01_qc/mosdepth/51-A_Spain_1810_2024_S51.mosdepth.region.dist.txt",
                "01_qc/mosdepth/51-A_Spain_1810_2024_S51.mosdepth.summary.txt",
                "01_qc/mosdepth/51-A_Spain_1810_2024_S51.per-base.bed.gz",
                "01_qc/mosdepth/51-A_Spain_1810_2024_S51.per-base.bed.gz.csi",
                "01_qc/mosdepth/51-A_Spain_1810_2024_S51.regions.bed.gz",
                "01_qc/mosdepth/51-A_Spain_1810_2024_S51.regions.bed.gz.csi",
                "02_host_removal",
                "02_host_removal/51-A_Spain_1810_2024_S51.host.sorted.bam",
                "02_host_removal/51-A_Spain_1810_2024_S51.host.sorted.bam.bai",
                "02_host_removal/stats",
                "02_host_removal/stats/51-A_Spain_1810_2024_S51.host.flagstat",
                "02_host_removal/stats/51-A_Spain_1810_2024_S51.host.idxstats",
                "02_host_removal/stats/51-A_Spain_1810_2024_S51.host.stats",
                "03_viral_reads",
                "03_viral_reads/51-A_Spain_1810_2024_S51.viral_1.fastq.gz",
                "03_viral_reads/51-A_Spain_1810_2024_S51.viral_2.fastq.gz",
                "06_alignment",
                "06_alignment/51-A_Spain_1810_2024_S51.bam",
                "06_alignment/51-A_Spain_1810_2024_S51.viral.sorted.bam",
                "06_alignment/51-A_Spain_1810_2024_S51.viral.sorted.bam.bai",
                "06_alignment/stats",
                "06_alignment/stats/51-A_Spain_1810_2024_S51.viral.flagstat",
                "06_alignment/stats/51-A_Spain_1810_2024_S51.viral.idxstats",
                "06_alignment/stats/51-A_Spain_1810_2024_S51.viral.stats",
                "08_consensus",
                "08_consensus/count_tables",
                "08_consensus/count_tables/51-A_Spain_1810_2024_S51.count_table.csv",
                "08_consensus/pileup",
                "08_consensus/pileup/51-A_Spain_1810_2024_S51.mpileup",
                "pipeline_info"
            ],
            [
                "viral_reference.merged.fasta:md5,887fe7e90382b03d3e9aabb2b03f2ede",
                "viral_reference.merged.fasta.fai:md5,c1207acbd8f4d0ce7d301a735c40b378",
                "51-A_Spain_1810_2024_S51.mosdepth.global.dist.txt:md5,38b598d0f43357acae7951963f14e7e6",
                "51-A_Spain_1810_2024_S51.mosdepth.region.dist.txt:md5,f756a1732644c0b72ac607c0326fbe9a",
                "51-A_Spain_1810_2024_S51.mosdepth.summary.txt:md5,02f1f6df2eb126d103778c565a75d997",
                "51-A_Spain_1810_2024_S51.per-base.bed.gz:md5,a6ac9688ad7c4d797978fcafb651b6cc",
                "51-A_Spain_1810_2024_S51.per-base.bed.gz.csi:md5,8a42178224c7c9c44659cceec68294a5",
                "51-A_Spain_1810_2024_S51.regions.bed.gz:md5,be22996bd2d56c17e689856d135c3d9b",
                "51-A_Spain_1810_2024_S51.regions.bed.gz.csi:md5,787a3d43a23aa2d593d2994c5e56aea8",
                "51-A_Spain_1810_2024_S51.host.sorted.bam:md5,0ccc97295ebd3a3f1f5902062aad22a2",
                "51-A_Spain_1810_2024_S51.host.sorted.bam.bai:md5,0b7739ae61977339522f05ea8be91cef",
                "51-A_Spain_1810_2024_S51.host.flagstat:md5,e2ab779358cbb754f62bcbf11d01e700",
                "51-A_Spain_1810_2024_S51.host.idxstats:md5,0cbe57910873f609f35fe6ffca9c0c71",
                "51-A_Spain_1810_2024_S51.host.stats:md5,43e55084a25f3eb12987cc8890500517",
                "51-A_Spain_1810_2024_S51.viral_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "51-A_Spain_1810_2024_S51.viral_2.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                "51-A_Spain_1810_2024_S51.bam:md5,edc8c8116f86570e3388597b3dbb184f",
                "51-A_Spain_1810_2024_S51.viral.sorted.bam:md5,e551d87d8fe334b93d533001f02bae42",
                "51-A_Spain_1810_2024_S51.viral.sorted.bam.bai:md5,aecd6c2fc2054bff0d0d89594f691632",
                "51-A_Spain_1810_2024_S51.viral.flagstat:md5,a6ab28b5e4272b95a510bbf64f570da2",
                "51-A_Spain_1810_2024_S51.viral.idxstats:md5,ca945141d602e706079efc8282f1e5f6",
                "51-A_Spain_1810_2024_S51.viral.stats:md5,09b9971223c7989f907ce9c04ae9aa1c",
                "51-A_Spain_1810_2024_S51.count_table.csv:md5,5f9138470ee32eef3059d920cf54b698",
                "51-A_Spain_1810_2024_S51.mpileup:md5,0b7db00a4044ac986b04e4353ee8d5be"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T11:40:35.257683"
    },
    "remove_host_illumina": {
        "content": [
            12,
            [
                "00_reference",
                "00_reference/viral_reference.merged.fasta",
                "00_reference/viral_reference.merged.fasta.fai",
                "02_host_removal",
                "02_host_removal/51-A_Spain_1810_2024_S51.host.sorted.bam",
                "02_host_removal/51-A_Spain_1810_2024_S51.host.sorted.bam.bai",
                "02_host_removal/stats",
                "02_host_removal/stats/51-A_Spain_1810_2024_S51.host.flagstat",
                "02_host_removal/stats/51-A_Spain_1810_2024_S51.host.idxstats",
                "02_host_removal/stats/51-A_Spain_1810_2024_S51.host.stats",
                "03_viral_reads",
                "03_viral_reads/51-A_Spain_1810_2024_S51.viral_1.fastq.gz",
                "03_viral_reads/51-A_Spain_1810_2024_S51.viral_2.fastq.gz",
                "pipeline_info"
            ],
            [
                "viral_reference.merged.fasta:md5,887fe7e90382b03d3e9aabb2b03f2ede",
                "viral_reference.merged.fasta.fai:md5,c1207acbd8f4d0ce7d301a735c40b378",
                "51-A_Spain_1810_2024_S51.host.sorted.bam:md5,0906ce95887672ec727188ab0a077bed",
                "51-A_Spain_1810_2024_S51.host.sorted.bam.bai:md5,3163f45f9b29e79a55e8af63fbc504dd",
                "51-A_Spain_1810_2024_S51.host.flagstat:md5,72b8602022c7a35067db07eefcc74592",
                "51-A_Spain_1810_2024_S51.host.idxstats:md5,b9e7d4b384ff996afa680d7706ef3498",
                "51-A_Spain_1810_2024_S51.host.stats:md5,f4bdbb91e08e2ca614fe7a8c47672a2b",
                "51-A_Spain_1810_2024_S51.viral_1.fastq.gz:md5,d2fd722c94dfff3e032c06f71e9017e1",
                "51-A_Spain_1810_2024_S51.viral_2.fastq.gz:md5,93e6c026d7d3ba2c3a2f4fdcae52754c"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T11:40:01.548087"
    }
}